node/src/chain_spec.rs

```diff
@@ -120,7 +126,7 @@ fn eden_testnet_genesis(
 
 		// Consensus
 		validators_set: ValidatorsSetConfig {
-			members: collators.iter().map(|x| x.0.clone()).collect::<Vec<_>>(),
+			members: BoundedVec::try_from(validator_members).expect("Couldbe Max Overflow"),
```

Comment:
```suggestion
			members: BoundedVec::try_from(validator_members).unwrap_or_default(),
```

Yes I think a default approach is a more defensive alternative, but again this should never happen.

Reply:
* In my view, this function is part of bootstrap sequence, so early detect of the error is good to make the corrective action early.

* Default is good suggestion but it leads to runtime error, may be landed up in no valid validatorset as part of POA. as a result may lead to no block production as latter part in the sequence.

* expect() or panic!() are not harm in this context, upstream template code is using it for checking the WASM binary.

![](https://i.imgur.com/mqpXZYV.png)

Reply:
I know I have seen it